cli/internal/api/client.go

```diff
@@ -12,26 +14,23 @@ import (
 const (
 	defaultAPIURL = "https://api.cloudquery.io"
 	envAPIURL     = "CLOUDQUERY_API_URL"
+	envCLIAPIURL  = "CLOUDQUERY_CLI_API_URL"
+	envCLIToken   = "CLOUDQUERY_CLI_TOKEN"
 )
 
+// NewClient creates a new client with the given token.
 func NewClient(token string) (*cloudquery_api.ClientWithResponses, error) {
-	c, err := cloudquery_api.NewClientWithResponses(env.GetEnvOrDefault(envAPIURL, defaultAPIURL),
-		cloudquery_api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
-			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
-			return nil
-		}))
-	if err != nil {
-		return nil, fmt.Errorf("failed to create api client: %w", err)
-	}
-	return c, nil
+	return newClient(token, false)
 }
 
+// NewAnonymousClient creates a client that doesn't require authentication.
 func NewAnonymousClient() (*cloudquery_api.ClientWithResponses, error) {
-	c, err := cloudquery_api.NewClientWithResponses(env.GetEnvOrDefault(envAPIURL, defaultAPIURL))
-	if err != nil {
-		return nil, fmt.Errorf("failed to create api client: %w", err)
-	}
-	return c, nil
+	return NewClient("")
+}
+
+// NewLocalClient creates a client that connects to the local API if possible. If not, it falls back to the regular API using `nonLocalToken`.
+func NewLocalClient(nonLocalToken string) (*cloudquery_api.ClientWithResponses, error) {
```

Comment:
Next step would be to go through cli client usage and replace some of the `NewClient` usage with `NewLocalClient` (such as sync, test conn ...)